cvat/apps/onepanelio/views.py

```diff
@@ -39,3 +95,248 @@ def get_workflow_templates(request):
+
+
+@api_view(['POST'])
+def create_annotation_model(request, pk):
+    """
+        Executes workflow selected by User.
+    """
+    
+    db_task = TaskModel.objects.get(pk=pk)
+    db_labels = db_task.label_set.prefetch_related('attributespec_set').all()
+    db_labels = {db_label.id:db_label.name for db_label in db_labels}
+    num_classes = len(db_labels.values())
+
+    slogger.glob.info("Creating annotation model for task: {} with num_classes {}".format(db_task.name,num_classes))
+
+    form_data = json.loads(request.body.decode('utf-8'))
+    slogger.glob.info("Form data without preprocessing {} {}".format(form_data, type(form_data)))
+ 
+    form_args = form_data['arguments']
+    time = datetime.now()
+    stamp = time.strftime("%m%d%Y%H%M%S")
+
+    if "cpu" in form_data['machine_type']:
+        tf_image = "tensorflow/tensorflow:1.13.1-py3"
+        machine ="Standard_D4s_v3"
+    else:
+        tf_image = "tensorflow/tensorflow:1.13.1-gpu-py3"
+        machine = "Standard_NC6"
+
+    list_of_args = form_args.split(';')
+    args_and_vals = {}
+    for i in list_of_args:
+        if i == "":
+            continue
+        arg = i.split('=')
+        args_and_vals[arg[0]] = arg[1]
+
+    if '--stage1_epochs' not in args_and_vals:
+        args_and_vals["--stage1_epochs"] = 1
+    if '--stage2_epochs' not in args_and_vals:
+        args_and_vals["--stage2_epochs"] = 2
+    if '--stage3_epochs' not in args_and_vals:
+        args_and_vals["--stage3_epochs"] = 3
+
+    cloud_prefix = os.getenv('ONEPANEL_RESOURCE_NAMESPACE')+ '/annotation-dump/'
+
+    # dump training data on cloud
+    bucket_name, dataset_name = dump_training_data(int(form_data['project_uid']), db_task, stamp, form_data['dump_format'], cloud_prefix)
+   
+    #execute workflow
+    configuration = onepanel_authorize()
+
+    # Enter a context with an instance of the API client
+    with onepanel.core.api.ApiClient(configuration) as api_client:
+        # Create an instance of the API class
+        api_instance = onepanel.core.api.WorkflowServiceApi(api_client)
+        namespace = os.getenv('ONEPANEL_RESOURCE_NAMESPACE') # str | 
+        params = []
+        # params.append(Parameter(name="source", value="https://github.com/onepanelio/Mask_RNN.git"))
+        params.append(Parameter(name="dataset-path", value=cloud_prefix+dataset_name))
+        params.append(Parameter(name="bucket-name", value=bucket_name))
+        params.append(Parameter(name='task-name', value=db_task.name))
+        # params.append(Parameter(name='num-classes', value=str(num_classes)))
+        params.append(Parameter(name='extras', value=json.dumps(args_and_vals).replace(" ","").replace("{","").replace("}","").replace(":","=")))
+        params.append(Parameter(name="tf-image", value=tf_image))
+        params.append(Parameter(name="sys-node-pool", value=machine))
+        if 'TFRecord' in form_data['dump_format']:
+            if "base_model" in form_data and "tfod" in form_data['base_model']:
```

Comment:
This if-else seems to be repeated in the else statement below.
I think we can pull this out to reduce code duplication.

Reply:
yes, in fact, this whole function will be replaced once we have different payload from the front-end.